specification/web/resource-manager/Microsoft.Web/stable/2022-09-01/StaticSites.json

```diff
@@ -3811,6 +4647,164 @@
+          "type": "string",
+          "readOnly": true
+        },
+        "type": {
+          "description": "The type of configuration file.",
+          "type": "string",
+          "readOnly": true
+        }
+      },
+      "x-ms-client-flatten": true
+    },
+    "StaticSiteDatabaseConnectionOverviewARMResource": {
+      "description": "Static Site Database Connection Overview ARM resource.",
+      "type": "object",
+      "allOf": [
+        {
+          "$ref": "./CommonDefinitions.json#/definitions/ProxyOnlyResource"
+        }
+      ],
+      "properties": {
+        "properties": {
+          "description": "StaticSiteDatabaseConnectionOverviewARMResource resource specific properties",
+          "type": "object",
+          "properties": {
+            "resourceId": {
+              "description": "The resource id of the database.",
+              "type": "string",
+              "readOnly": true
+            },
+            "connectionIdentity": {
+              "description": "If present, the identity is used in conjunction with ConnectionString to connect to the database.",
+              "type": "string",
+              "readOnly": true
+            },
+            "connectionString": {
+              "description": "The connection string to use to connect to the database.",
+              "type": "string",
+              "readOnly": true
+            },
+            "region": {
+              "description": "The region of the database resource.",
+              "type": "string",
+              "readOnly": true
+            },
+            "configurationFiles": {
+              "description": "A list of configuration files associated with this database connection.",
+              "type": "array",
+              "items": {
+                "$ref": "#/definitions/StaticSiteDatabaseConnectionConfigurationFileOverview"
+              },
+              "x-ms-identifiers": [],
+              "readOnly": true
+            },
+            "name": {
+              "description": "If present, the name of this database connection resource.",
+              "type": "string",
+              "readOnly": true
+            }
+          },
+          "x-ms-client-flatten": true
+        }
+      }
+    },
+    "StaticSiteDatabaseConnectionOverviewCollection": {
+      "description": "Collection of static site database connections.",
+      "required": [
+        "value"
+      ],
+      "type": "object",
+      "properties": {
+        "value": {
+          "description": "Collection of resources.",
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/StaticSiteDatabaseConnectionOverviewARMResource"
+          }
+        },
+        "nextLink": {
+          "description": "Link to next page of resources.",
+          "type": "string",
+          "readOnly": true
+        }
+      }
+    },
+    "StaticSiteDatabaseConnectionPatchRequestPropertiesARMResource": {
```

Comment:
>StaticSiteDatabaseConnectionPatchRequestPropertiesARMResource [](http://example.com/codeflow?start=5&length=61)

Why did you remove the allOf inheritance of ProxyResoruce?

Reply:
The ProxyOnlyResource definition contains the properties `name` and `type` and I was getting a flagged with this error: https://github.com/Azure/azure-openapi-validator/blob/main/docs/un-supported-patch-properties.md

Reply:
Oh OK. That seems weird. I mean, I thought they are all modeled as read-only, aren't they?

Reply:
It might have been a problem with the specific ProxyResource definition that you were using.

          "$ref": "./CommonDefinitions.json#/definitions/ProxyOnlyResource"

Reply:
I think this is fine cause you really should not be providing those top level properties for PATCH